File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) -1
        maxWater = 0

        while left< right:
            h = min(heights[left], heights[right])
            w = right - left
            area= h*w

            maxWater = max(maxWater, area)

            if heights[left] < heights[right]:
                left += 1
            else:
                right -=1

        return maxWater

File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int left = matrix.size();
        int right = matrix[0].size();
        int r = 0, c = right - 1;

        while(r < left && c>=0){
            if(matrix[r][c] > target){
                c--;
            } else if(matrix[r][c]< target){
                r++;
            } else {
                return true;
            }
        }
        return false;
    }
};
